### PR TITLE
Expose IPCChannel from the framework and SDK - Closes #5186

### LINKED ISSUE #5186: Expose IPCChannel from the framework and SDK
### Description

Rename ChildProcessChannel to `IPCChannel` and expose from `lisk-framework` and `lisk-sdk`.

### Motivation

The actions and events should be accessible through this IPC channel, so that application built with the `lisk-sdk` can be accessible through this IPC channel outside of the application.

### Acceptance Criteria

- IPCChannel are exposed from the `lisk-sdk` and `lisk-framework`, so that libraries user can utilise them

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?

This PR resolves #5186

### How was it solved?

Renamed the files and names from `ChildProcessChannel` to `IPCChannel`

### How was it tested?

Run all tests

### CODE REVIEW

File: framework/test/integration/specs/controller/ipc_channel.spec.ts [hunk 1]
```diff
@@ -67,24 +67,24 @@ const beta = {
 	},
 };
 /* eslint-disable jest/no-disabled-tests */
-describe.skip('ChildProcessChannel', () => {
+describe.skip('IPCChannel', () => {
```
Comment (ManuGowda) [inline framework/test/integration/specs/controller/ipc_channel.spec.ts:70]: skip?

File: framework/src/index.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ export { Application } from './application';
 export { version } from './version';
 export { systemDirs } from './application/system_dirs';
 export { BasePlugin, PluginInfo } from './plugins/base_plugin';
+export { IPCChannel } from './controller/channels';
 export type { BaseChannel } from './controller/channels';
 export type { EventsArray } from './controller/event';
```
Comment (ManuGowda) [inline framework/src/index.ts:21]: maybe expose these from `controller/index.ts` and `export {} from './controller'`

File: framework/test/integration/specs/controller/ipc_channel.spec.ts (hunk 1 above)
Comment (nazarhussain) [reply framework/test/integration/specs/controller/ipc_channel.spec.ts:70]: Yes skipped in recent PR, have another issue to look into why they are unstable.

File: framework/src/index.ts (hunk 2 above)
Comment (nazarhussain) [reply framework/src/index.ts:21]: These are exposed at `controller/index.ts` but we need only `IPCChannel` channel to expose from framework.

File: framework/test/integration/specs/controller/ipc_channel.spec.ts (hunk 1 above)
Comment (nazarhussain) [reply framework/test/integration/specs/controller/ipc_channel.spec.ts:70]: https://github.com/LiskHQ/lisk-sdk/issues/5512

**Pull request merged**